### DISCUSSION TITLE
Site should redirect http:// to https://

### DISCUSSION THREAD

**Discussion by coliff** [RetroAchievements/RAWeb#840] [Ideas]
Currently the site is available over both HTTP and HTTPS. It'd be good for security and performance if it always redirected to HTTPS. Ideally this should be done on the server-side config... but as a fallback a simple script in the head could work.
```html
<script>window.location.protocol!='https:'&&(location.href=location.href.replace("http://","https://"))</script>
```

**Comment by Sanaki** [comment]
If done, it has to be done very carefully. RetroArch is used on a number of devices that may not even support https. We don't want to cut off achievements on those devices if we can avoid it.

**Comment by luchaos** [comment]
Due to the clients integration we cannot safely do that without locking them out.
V2 will split site requests from API requests where we have fine grained control over such thing. Site and Web API will only be available via https - the emulator/connect API will continue to allow both.
The same applies when we were thinking about enabling HSTS for the domain - which would be a disaster, actually.

Now that i think of it - a frontend solution with javascript might actually be the easiest solution for that in V1 🤔

**Comment by ScottFromDerby** [comment]
PS Vita for instance is a popular choice for getting achievements via RetroArch (myself included), and the Vita has no https support by default iirc.

**Reply by televandalist** [reply]
#vitagang ❤️

**Comment by Flashfire42** [comment]
Perhaps have it serve forced HTTPS per user agent? Its been a while since I messed with anything website code related but it could be possible
